### PR TITLE
[EAGLE-2092] Test for  custom code operator

### PR DESCRIPTION
**Pull request by marco-clarifai**

### CODE REVIEW

File: tests/client/test_custom_code_operators.py [hunk 1]
```diff
@@ -0,0 +1,79 @@
...
+
+    for i in range(0, MAX_RETRY_ATTEMPTS):
+        resp = stub.GetModel(
+            service_pb2.GetModelRequest(
+                model_id=model_id,
+            ),
+            metadata=metadata(),
+        )
+        latest_status_code = resp.model.model_version.status.code
+        if latest_status_code != status_code_pb2.MODEL_TRAINED:
```
Comment (moldovan-clarifai) [inline tests/client/test_custom_code_operators.py:47]: we should break when model is trained

File: tests/client/test_custom_code_operators.py [hunk 2]
```diff
@@ -0,0 +1,79 @@
...
+            time.sleep(2)
+
+    assert latest_status_code == status_code_pb2.MODEL_TRAINED
+
+    inputs = [
+        resources_pb2.Input(
+            id="321",
```
Comment (moldovan-clarifai) [inline tests/client/test_custom_code_operators.py:54]: why do u provide ID? is this required?

**Review by moldovan-clarifai — DISMISSED** [review]

File: tests/client/test_custom_code_operators.py [hunk 3]
```diff
@@ -0,0 +1,79 @@
...
+    output_info = resources_pb2.OutputInfo(params=output_info_params)
+    model = resources_pb2.Model(
+        id=model_id,
+        model_type_id="custom-code-operator",
+        output_info=output_info,
+    )
+    req = service_pb2.PostModelsRequest(model=model)
+    raise_on_failure(stub.PostModels(req, metadata=metadata()))
+
+    for i in range(0, MAX_RETRY_ATTEMPTS):
```
Comment (HarmitMinhas96) [inline tests/client/test_custom_code_operators.py:39]: We have a helper for this: wait_for_model_trained

Comment (marco-clarifai) [reply tests/client/test_custom_code_operators.py:39]: The status we assign to the operator is MODEL_UPLOADING, not MODEL_TRAINING or MODEL_QUEUED_FOR_TRAINING.

File: tests/client/test_custom_code_operators.py (hunk 2 above)
Comment (marco-clarifai) [reply tests/client/test_custom_code_operators.py:54]: I mixed up the implementation a bit. Only the operator code needs to handle the ids, not the client code. Removed.

File: tests/client/test_custom_code_operators.py (hunk 1 above)
Comment (marco-clarifai) [reply tests/client/test_custom_code_operators.py:47]: Fixed.

File: tests/client/test_custom_code_operators.py (hunk 3 above)
Comment (HarmitMinhas96) [reply tests/client/test_custom_code_operators.py:74]: Ah, right

File: tests/client/test_custom_code_operators.py [hunk 4]
```diff
@@ -0,0 +1,113 @@
...
+    Add custom code operator model, model version, run a predict, delete CCO.
+    """
+    stub = service_pb2_grpc.V2Stub(channel)
+    model_id = "coperator_" + uuid.uuid4().hex[:20]
+
+    model = resources_pb2.Model(
+        id=model_id,
+        model_type_id="custom-code-operator",
+    )
+    req = service_pb2.PostModelsRequest(model=model)
```
Comment (HarmitMinhas96) [inline tests/client/test_custom_code_operators.py:61]: Thanks for using the new way!

**Review by HarmitMinhas96 — APPROVED** [review]

File: tests/client/test_custom_code_operators.py (hunk 4 above)
Comment (marco-clarifai) [reply tests/client/test_custom_code_operators.py:61]: Thanks, I almost forgot!

**Pull request merged**